proxy/src/avatar.rs

```diff
@@ -116,34 +123,35 @@ impl Color {
 
     /// Compute the lightness of a color.
     #[must_use]
-    pub fn lightness(&self) -> f32 {
-        let r = self.r as f32;
-        let g = self.g as f32;
-        let b = self.b as f32;
-        let n = 0xff as f32;
+    pub fn lightness(self) -> f32 {
+        let r = f32::from(self.r);
+        let g = f32::from(self.g);
+        let b = f32::from(self.b);
```

Comment:
`f32::from` is implemented as `x as f32`, so this is just a more verbose way of saying the same thing.

Reply:
Not quite:

>  Rust’s as keyword will perform many kinds of conversions, including silently lossy conversions. Conversion functions such as i32::from will only perform lossless conversions. Using the conversion functions prevents conversions from turning into silent lossy conversions if the types of the input expressions ever change, and make it easier for people reading the code to know that the conversion is lossless.

It might be a bit pedantic in this case, but generally avoiding lossy conversions will make the code safer and correct.

Reply:
Yeah I meant for `u8` sorry, not for all cases. In the case of `u8 -> f32` the conversion is lossless.

Reply:
That's clear, it's what the comment above hints at that if the code evolves and the `u8` will become something else having the explicit `from` will surface that there is a transition from a lossless conversion to a lossy one, so it's about safety over time.